arch/arm/neon_intrins.h

```diff
@@ -36,20 +36,34 @@
 #  ifndef ARM_NEON_HASLD4
 
 static inline uint16x8x4_t vld1q_u16_x4(uint16_t const *a) {
+#if defined(_MSC_VER)
+#  pragma warning( push )
+#  pragma warning( disable: 4204)
+#endif
     uint16x8x4_t ret = (uint16x8x4_t) {{
                           vld1q_u16(a),
                           vld1q_u16(a+8),
                           vld1q_u16(a+16),
                           vld1q_u16(a+24)}};
+#if defined(_MSC_VER)
+#  pragma warning( pop )
+#endif
     return ret;
 }
 
 static inline uint8x16x4_t vld1q_u8_x4(uint8_t const *a) {
+#if defined(_MSC_VER)
+#  pragma warning( push )
+#  pragma warning( disable: 4204)
+#endif
     uint8x16x4_t ret = (uint8x16x4_t) {{
```

Comment:
Why not do:
```c
uint8x16x4_t ret;
ret.val[0] = vld1q_u8(a);
ret.val[1] = vld1q_u8(a+16);
ret.val[2] = vld1q_u8(a+32);
ret.val[3] = vld1q_u8(a+48);
```
It is fully compatible and no need for pragma stuff.

Reply:
I thought about that, but wasn't sure if it compiles to equally fast code... At least optimizer in gcc used to have problems with setting individual elements of vector with array notation.